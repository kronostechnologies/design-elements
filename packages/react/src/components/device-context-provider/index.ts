export { DeviceContextProvider, useDeviceContext } from './device-context-provider';
export type { DeviceContextProviderProps, DeviceType } from './device-context-provider';
