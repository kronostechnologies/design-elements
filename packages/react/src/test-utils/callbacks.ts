export function doNothing(): void {
    // do nothing
}
