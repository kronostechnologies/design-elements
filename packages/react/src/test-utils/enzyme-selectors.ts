import { ReactWrapper, ShallowWrapper } from 'enzyme';

enum ModifierEnum {
    StartsWith = '^',
    EndsWith = '$',
    ContainsOptionalHyphen = '|',
    Contains = '*',
    ContainsWord = '~',
    None = '',
}
type SelectorModifier = `${ModifierEnum}`;

function filterHtmlNodesOnly<W extends ReactWrapper | ShallowWrapper>(
    nodes: W,
): W {
    return nodes
        .hostNodes()
        .filterWhere((x) => x.name() !== null && !x.name().startsWith('styled.')) as W;
}

export function findByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node over the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    const modifier = options?.modifier ?? '';

    let nodes = wrapper.find(`[data-testid${modifier}="${testId}"]`) as W;

    if (options?.nodeName) {
        nodes = nodes.filterWhere((x) => x.name() === options.nodeName) as W;
    }
    if (options?.htmlNodesOnly) {
        nodes = filterHtmlNodesOnly(nodes);
    }

    return nodes;
}

export function getByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node instead of the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    return findByTestId(wrapper, testId, options).at(0) as W;
}
