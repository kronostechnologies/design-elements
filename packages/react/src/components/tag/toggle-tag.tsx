import { forwardRef, MouseEventHandler, Ref, SVGProps, useCallback, useState } from 'react';
import styled from 'styled-components';
import { focus } from '../../utils/css-state';
import { useDeviceContext } from '../device-context-provider/device-context-provider';
import { Icon, IconName } from '../icon/icon';

export type ToggleTagSize =
    | 'small'
    | 'medium';

export interface ToggleTagValue {
    id?: string;
    label: string;
}

export interface BaseTagProps {
    className?: string;
    size?: ToggleTagSize;
    value: ToggleTagValue;
    iconName?: IconName;
}

export interface BaseTagStylingProps {
    $isMobile: boolean;
    $tagSize: ToggleTagSize;
    $hasIcon: boolean;
}

export interface ToggleTagProps extends BaseTagProps {
    onClick?(tag: ToggleTagValue): void;
    /**
     * Whether the tag is selected.
     * Can manually set the selected state of the tag, without the need for a click event.
     */
    selected?: boolean;
}

export interface ToggleTagStylingProps extends BaseTagStylingProps {
    $selected?: boolean;
}

export function getFontSize({ $isMobile }: BaseTagStylingProps): number {
    return $isMobile ? 0.875 : 0.75;
}

export function getIconSize(isMobile: boolean): string {
    return isMobile ? '20' : '12';
}

export function isMedium(tagSize: ToggleTagSize): tagSize is 'medium' {
    return tagSize === 'medium';
}

export function isSmall(tagSize: ToggleTagSize): tagSize is 'small' {
    return tagSize === 'small';
}

export function getPadding({ $isMobile, $tagSize }: BaseTagStylingProps): string {
    return $isMobile || isMedium($tagSize) ? '0 var(--spacing-1x)' : '0 var(--spacing-half)';
}

export function getLineHeight({ $isMobile, $tagSize }: BaseTagStylingProps): number {
    if ($isMobile) {
        return isSmall($tagSize) ? 1.5 : 1.875;
    }
    return isSmall($tagSize) ? 1 : 1.5;
}

const TagContainer = styled.button<ToggleTagStylingProps>`
    align-items: center;
    background-color: ${({ theme, $selected }) => ($selected ? theme.component['tag-selected-background-color'] : theme.component['tag-background-color'])};
    border: 1px solid ${({ theme, $selected }) => ($selected ? theme.component['tag-selected-border-color'] : theme.component['tag-border-color'])};
    border-radius: ${({ $tagSize }) => (isSmall($tagSize) ? 'var(--border-radius-2halfx)' : 'var(--border-radius-3x)')};
    display: inline-flex;
    font-family: var(--font-family);
    font-weight: ${({ $selected }) => ($selected ? '600' : '400')};
    padding: ${getPadding};
    padding-left: ${({ $tagSize }) => (isSmall($tagSize) ? 'var(--spacing-1x)' : 'var(--spacing-1halfx)')};
    padding-right: ${({ $tagSize }) => (isSmall($tagSize) ? 'var(--spacing-1x)' : 'var(--spacing-1halfx)')};

    & + & {
        margin-left: var(--spacing-1x);
    }

    ${focus};

    &:hover {
        background-color: ${({ theme, $selected }) => ($selected ? theme.component['tag-selected-background-color'] : theme.component['tag-hover-background-color'])};
        border-color: ${({ theme, $selected }) => ($selected ? theme.component['tag-selected-border-color'] : theme.component['tag-hover-border-color'])};
    }
`;

const TagLabel = styled.span<ToggleTagStylingProps>`
    color: ${({ theme, $selected }) => ($selected ? theme.component['tag-selected-text-color'] : theme.component['tag-text-color'])};
    display: inline-block;
    font-size: ${getFontSize}rem;
    line-height: ${getLineHeight}rem;
    max-width: 312px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;

    &:hover {
        color: ${({ $selected, theme }) => ($selected ? theme.component['tag-selected-text-color'] : theme.component['tag-hover-text-color'])};
    }
`;

const StyledIcon = styled(Icon)<SVGProps<SVGSVGElement> & ToggleTagStylingProps>`
    color: ${({ theme, $selected }) => ($selected ? theme.component['tag-selected-icon-color'] : theme.component['tag-icon-color'])};
    height: var(--size-1x);
    margin-right: var(--spacing-half);
    vertical-align: text-bottom;
    width: var(--size-1x);

    &:hover {
        color: ${({ $selected, theme }) => ($selected ? theme.component['tag-selected-icon-color'] : theme.component['tag-hover-icon-color'])};
    }
`;

export const ToggleTag = forwardRef(({
    className,
    iconName,
    size = 'medium',
    value,
    selected = undefined,
    onClick,
}: ToggleTagProps, ref: Ref<HTMLButtonElement>) => {
    const { isMobile } = useDeviceContext();

    const hasIcon = !!iconName;
    const hasIconLabel = !(value.label.toLowerCase() === iconName?.toLowerCase());
    const [isSelectedInternal, setSelectedInternal] = useState(selected ?? false);
    const isSelected = selected !== undefined ? selected : isSelectedInternal;

    const handleClick: MouseEventHandler = useCallback(() => {
        if (selected === undefined) {
            setSelectedInternal(!isSelectedInternal);
        }
        onClick?.(value);
    }, [isSelectedInternal, onClick, selected, value]);

    return (
        <TagContainer
            ref={ref}
            className={className}
            onClick={handleClick}
            $isMobile={isMobile}
            $tagSize={size}
            $hasIcon={hasIcon}
            $selected={isSelected}
        >
            {hasIcon && (
                <StyledIcon
                    aria-label={hasIconLabel ? iconName : undefined}
                    aria-hidden={!hasIconLabel}
                    data-testid={`${value.label}-icon`}
                    name={iconName}
                    size={getIconSize(isMobile)}
                    role="img"
                    $isMobile={isMobile}
                    $tagSize={size}
                    focusable
                    $selected={isSelected}
                    $hasIcon={hasIcon}
                />
            )}

            <TagLabel
                $isMobile={isMobile}
                $tagSize={size}
                $selected={isSelected}
                $hasIcon={hasIcon}
            >
                {value.label}
            </TagLabel>
        </TagContainer>
    );
});

ToggleTag.displayName = 'ToggleTag';
