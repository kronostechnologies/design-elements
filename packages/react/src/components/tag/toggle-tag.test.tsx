import { shallow } from 'enzyme';
import { getByTestId } from '../../test-utils/enzyme-selectors';
import { renderWithProviders } from '../../test-utils/renderer';
import { DeviceType } from '../device-context-provider/device-context-provider';
import { ToggleTag, ToggleTagProps } from './toggle-tag';

describe('ToggleTag', () => {
    const tagSizes: ToggleTagProps['size'][] = ['small', 'medium'];
    tagSizes.forEach((size) => {
        describe(`ToggleTag ${size}`, () => {
            it('should call onClick callback when tag is clicked', () => {
                const callback = jest.fn();
                const wrapper = shallow(
                    <ToggleTag size={size} value={{ label: 'Test' }} onClick={callback} />,
                );

                wrapper.simulate('click');

                expect(callback).toHaveBeenCalledTimes(1);
            });

            it(
                'should have aria-hidden="false" and an aria-label on icon when the label is not the same as iconName',
                () => {
                    const wrapper = shallow(
                        <ToggleTag size={size} iconName="home" value={{ label: 'Test' }} />,
                    );

                    const testIconWrapper = getByTestId(wrapper, 'Test-icon');
                    expect(testIconWrapper.prop('aria-label')).toBe('home');
                    expect(testIconWrapper.prop('aria-hidden')).toBe(false);
                },
            );

            it(
                'should have aria-hidden="true" and no label on icon when the label is the same as the iconName',
                () => {
                    const wrapper = shallow(
                        <ToggleTag size={size} iconName="home" value={{ label: 'Home' }} />,
                    );

                    const testIconWrapper = getByTestId(wrapper, 'Home-icon');
                    expect(testIconWrapper.prop('aria-label')).toBe(undefined);
                    expect(testIconWrapper.prop('aria-hidden')).toBe(true);
                },
            );

            (['mobile', 'desktop'] as DeviceType[]).forEach((deviceType) => {
                it(`matches snapshot (${deviceType})`, () => {
                    const tree = renderWithProviders(<ToggleTag size={size} value={{ label: 'Test' }} />, 'desktop');

                    expect(tree).toMatchSnapshot();
                });

                it(`matches snapshot (${deviceType})`, () => {
                    const tree = renderWithProviders(<ToggleTag size={size} value={{ label: 'Test' }} />, 'mobile');

                    expect(tree).toMatchSnapshot();
                });

                it(`matches snapshot (${deviceType} with icons)`, () => {
                    const tree = renderWithProviders(
                        <ToggleTag
                            size={size}
                            iconName="home"
                            value={{ label: 'Test' }}
                        />,
                    );

                    expect(tree).toMatchSnapshot();
                });

                it(`matches snapshot (${deviceType} clickable)`, () => {
                    const tree = renderWithProviders(
                        <ToggleTag
                            size={size}
                            value={{ label: 'Test' }}
                            onClick={jest.fn()}
                        />,
                    );

                    expect(tree).toMatchSnapshot();
                });
            });
        });
    });
});
