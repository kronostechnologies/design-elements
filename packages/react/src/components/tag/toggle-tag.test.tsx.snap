// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ToggleTag ToggleTag medium matches snapshot (desktop clickable) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-3x);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-1x);
  padding-left: var(--spacing-1halfx);
  padding-right: var(--spacing-1halfx);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1.5rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag medium matches snapshot (desktop with icons) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-3x);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-1x);
  padding-left: var(--spacing-1halfx);
  padding-right: var(--spacing-1halfx);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c3 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1.5rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c3:hover {
  color: #000000;
}

.c2 {
  color: #1B1C1E;
  height: var(--size-1x);
  margin-right: var(--spacing-half);
  vertical-align: text-bottom;
  width: var(--size-1x);
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <svg
    aria-hidden="false"
    aria-label="home"
    class="c2"
    color="currentColor"
    data-testid="Test-icon"
    focusable="true"
    height="12"
    role="img"
    width="12"
  />
  <span
    class="c3"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag medium matches snapshot (desktop) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-3x);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-1x);
  padding-left: var(--spacing-1halfx);
  padding-right: var(--spacing-1halfx);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1.5rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag medium matches snapshot (desktop) 2`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-3x);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-1x);
  padding-left: var(--spacing-1halfx);
  padding-right: var(--spacing-1halfx);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.875rem;
  line-height: 1.875rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag medium matches snapshot (mobile clickable) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-3x);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-1x);
  padding-left: var(--spacing-1halfx);
  padding-right: var(--spacing-1halfx);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1.5rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag medium matches snapshot (mobile with icons) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-3x);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-1x);
  padding-left: var(--spacing-1halfx);
  padding-right: var(--spacing-1halfx);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c3 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1.5rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c3:hover {
  color: #000000;
}

.c2 {
  color: #1B1C1E;
  height: var(--size-1x);
  margin-right: var(--spacing-half);
  vertical-align: text-bottom;
  width: var(--size-1x);
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <svg
    aria-hidden="false"
    aria-label="home"
    class="c2"
    color="currentColor"
    data-testid="Test-icon"
    focusable="true"
    height="12"
    role="img"
    width="12"
  />
  <span
    class="c3"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag medium matches snapshot (mobile) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-3x);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-1x);
  padding-left: var(--spacing-1halfx);
  padding-right: var(--spacing-1halfx);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1.5rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag medium matches snapshot (mobile) 2`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-3x);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-1x);
  padding-left: var(--spacing-1halfx);
  padding-right: var(--spacing-1halfx);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.875rem;
  line-height: 1.875rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag small matches snapshot (desktop clickable) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-2halfx);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-half);
  padding-left: var(--spacing-1x);
  padding-right: var(--spacing-1x);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag small matches snapshot (desktop with icons) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-2halfx);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-half);
  padding-left: var(--spacing-1x);
  padding-right: var(--spacing-1x);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c3 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c3:hover {
  color: #000000;
}

.c2 {
  color: #1B1C1E;
  height: var(--size-1x);
  margin-right: var(--spacing-half);
  vertical-align: text-bottom;
  width: var(--size-1x);
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <svg
    aria-hidden="false"
    aria-label="home"
    class="c2"
    color="currentColor"
    data-testid="Test-icon"
    focusable="true"
    height="12"
    role="img"
    width="12"
  />
  <span
    class="c3"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag small matches snapshot (desktop) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-2halfx);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-half);
  padding-left: var(--spacing-1x);
  padding-right: var(--spacing-1x);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag small matches snapshot (desktop) 2`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-2halfx);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-1x);
  padding-left: var(--spacing-1x);
  padding-right: var(--spacing-1x);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.875rem;
  line-height: 1.5rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag small matches snapshot (mobile clickable) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-2halfx);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-half);
  padding-left: var(--spacing-1x);
  padding-right: var(--spacing-1x);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag small matches snapshot (mobile with icons) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-2halfx);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-half);
  padding-left: var(--spacing-1x);
  padding-right: var(--spacing-1x);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c3 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c3:hover {
  color: #000000;
}

.c2 {
  color: #1B1C1E;
  height: var(--size-1x);
  margin-right: var(--spacing-half);
  vertical-align: text-bottom;
  width: var(--size-1x);
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <svg
    aria-hidden="false"
    aria-label="home"
    class="c2"
    color="currentColor"
    data-testid="Test-icon"
    focusable="true"
    height="12"
    role="img"
    width="12"
  />
  <span
    class="c3"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag small matches snapshot (mobile) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-2halfx);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-half);
  padding-left: var(--spacing-1x);
  padding-right: var(--spacing-1x);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.75rem;
  line-height: 1rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;

exports[`ToggleTag ToggleTag small matches snapshot (mobile) 2`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F1F2F2;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius-2halfx);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  font-weight: 400;
  padding: 0 var(--spacing-1x);
  padding-left: var(--spacing-1x);
  padding-right: var(--spacing-1x);
}

.c0 + .c0 {
  margin-left: var(--spacing-1x);
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover {
  background-color: #DBDEE1;
  border-color: #000000;
}

.c2 {
  color: #1B1C1E;
  display: inline-block;
  font-size: 0.875rem;
  line-height: 1.5rem;
  max-width: 312px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c2:hover {
  color: #000000;
}

<button
  class="c0 c1"
>
  <span
    class="c2"
  >
    Test
  </span>
</button>
`;
